src/api/app/models/user.rb

```diff
@@ -53,14 +53,62 @@ class User < ActiveRecord::Base
   # users have 0..1 user_registration records assigned to them
   has_one :user_registration
 
-  # This method returns an array with the names of all available
-  # password hash types supported by this User class.
-  def self.password_hash_types
-    default_password_hash_types
-  end
+  scope :all_without_nobody, -> { where("login != ?", nobody_login) }
 
-  after_create :create_home_project
+  # Add accessors for "new_password" property. This boolean property is set
+  # to true when the password has been set and validation on this password is
+  # required.
+  attr_accessor :new_password
+
+  # Generate accessors for the password confirmation property.
+  attr_accessor :password_confirmation
+
+  validates_presence_of :login, :email, :password, :password_hash_type, :state,
+                        :message => 'must be given'
+
+  validates_uniqueness_of :login,
+                          :message => 'is the name of an already existing user.'
+
+  validates_format_of :login,
+                      :with => %r{\A[\w \$\^\-\.#\*\+&'"]*\z},
+                      :message => 'must not contain invalid characters.'
+  validates_length_of :login,
+                      :in => 2..100, :allow_nil => true,
+                      :too_long => 'must have less than 100 characters.',
+                      :too_short => 'must have more than two characters.'
+
+  # We want a valid email address. Note that the checking done here is very
+  # rough. Email adresses are hard to validate now domain names may include
+  # language specific characters and user names can be about anything anyway.
+  # However, this is not *so* bad since users have to answer on their email
+  # to confirm their registration.
+  validates_format_of :email,
+                      :with => %r{\A([\w\-\.\#\$%&!?*\'\+=(){}|~]+)@([0-9a-zA-Z\-\.\#\$%&!?*\'=(){}|~]+)+\z},
+                      :message => 'must be a valid email address.'
+
+  # We want to validate the format of the password and only allow alphanumeric
+  # and some punctiation characters.
+  # The format must only be checked if the password has been set and the record
+  # has not been stored yet and it has actually been set at all. Make sure you
+  # include this condition in your :if parameter to validates_format_of when
+  # overriding the password format validation.
```

Comment:
@hennevogel Would you mind fi I remove the last sentence at all? I wouldn't know when we ever overwrite the if parameter...
Same for the validates_length_of, line 103